onnxruntime/core/graph/function.cc

```diff
@@ -44,12 +51,21 @@ void IOTypeConstraintHelper(const ONNX_NAMESPACE::FunctionProto& onnx_func_proto
       auto iter = input_name_idx_map.find(in_name);
       if (iter != input_name_idx_map.end()) {
         int idx = iter->second;
-        const auto& p = node_op_schema->inputs().at(i);
-        std::string type_str = p.GetTypeStr() + "in" + std::to_string(i);
+        std::string type_str = node_op_schema ? node_op_schema->inputs().at(i).GetTypeStr() + "in" + std::to_string(i) : "Tin" + std::to_string(i);
         input_types_list[idx] = std::make_pair(in_name, type_str);
         if (!type_constraint_map.count(type_str)) {
-          for (auto s : p.GetTypes()) {
-            type_constraint_map[type_str].emplace_back(*s);
+          // If schema is available for the node then get the allowed types from the schema 
```

Comment:
For now, this is okay. But, in the long run, this whole iteration through the function-body seems unnecessary. It doesn't really serve any purpose. It seems better to generalize the inference implementation to support an unconstrained type-variable, which would allow us to skip all this. May be worth adding a TODO for cleanup later on.

Reply:
Agree.. we should clean this up. Right now for schema less functions I add all type as supported. We can just apply this to all cases since there is no gain in walking through every node...